fdbserver/OnDemandStore.actor.cpp

```diff
@@ -68,11 +69,13 @@ void OnDemandStore::dispose() {
 	if (store) {
 		store->dispose();
 		store = nullptr;
+		err = Promise<Future<Void>>();
```

Comment:
What's the purpose of this statement?

Reply:
This allows us to reopen one of those. I had one test where this is happening. Without resetting err, `open` will fail with an assertion.

Reply:
Could you just do `err.reset()` instead? Then you wouldn't need to include `flow.h`.